packages/browser/__tests__/Session.spec.ts

```diff
@@ -271,37 +297,33 @@ describe("Session", () => {
       expect(mySession.info.sessionId).toEqual("mySession");
     });
 
-    function sleep(ms: number): Promise<void> {
-      return new Promise((resolve) => setTimeout(resolve, ms));
-    }
-
     it("prevents from hitting the token endpoint twice with the same auth code", async () => {
       const clientAuthentication = mockClientAuthentication();
       const obtainedSession: ISessionInfo = {
         isLoggedIn: true,
         sessionId: "mySession",
       };
-      let secondRequestIssued = false;
+      let continueAfterSecondRequest: (value?: unknown) => void;
+      const blockingResponse = new Promise((resolve, _reject) => {
+        continueAfterSecondRequest = resolve;
+      });
       const blockingRequest = async (): Promise<ISessionInfo> => {
-        while (!secondRequestIssued) {
-          // eslint-disable-next-line no-await-in-loop
-          await sleep(100);
-        }
+        await blockingResponse;
         return obtainedSession;
       };
       // The ClientAuthn's handleIncomingRedirect will only return when the
       // second Session's handleIncomingRedirect has been called.
-      clientAuthentication.handleIncomingRedirect = jest.fn(
-        async (_url: string) => blockingRequest()
-      );
+      clientAuthentication.handleIncomingRedirect = jest.fn(blockingRequest);
       const mySession = new Session({ clientAuthentication });
       const firstTokenRequest = mySession.handleIncomingRedirect(
         "https://my.app/?code=someCode&state=arizona"
       );
       const secondTokenRequest = mySession.handleIncomingRedirect(
         "https://my.app/?code=someCode&state=arizona"
       );
-      secondRequestIssued = true;
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      // We know that it has been set by the call to `blockingRequest`.
+      continueAfterSecondRequest!();
```

Comment:
I'm not sure why there are still Check warnings here, but I think they're from an earlier CI run, noting the ESLint comment on line 324.

Reply:
I'm not sure GH Actions even consider ESLint comments, which is kind of annoying, in particular since the repeat FOR EACH OS OF THE RUN MATRIX

Reply:
I mean, would you get the point if it was listed less than six times?

(But yeah, as far as I know the annotations arise from warnings thrown during the build commands that we specify, i.e. the specific ESLint command, so I think it's _supposed_ to consider them.)